### PR TITLE
#134 suppress exceptions in parsing `Fqsen` expressions for documented property types

### LINKED ISSUE #134: Do not crash when `Fqsen` parsing fails
As discussed with @sebastianbergmann (https://github.com/sebastianbergmann/phpunit/issues/3718), type annotations like `array<foo, bar>` and other more complex definitions currently lead to the phpdocumentor tooling to throw exceptions.

While it would be good to support these in full, this currently prevents picking up this tool at all in downstream consumers.

The suggestion is therefore to
 1. [x] Suppress `Fqsen` parsing errors
 2. [x] ~~Improve upstream `Fqsen` parsing tooling~~ not needed: see https://github.com/Roave/BackwardCompatibilityCheck/issues/143
 3. [x] ~~Adapt covariance/contravariance checks to support more complex types~~ moved to https://github.com/Roave/BackwardCompatibilityCheck/issues/143
 4. [x] ~~Extract covariance/contravariance checks to a new namespace/library that only does that, making it easier to upgrade its capabilities~~ moved to https://github.com/Roave/BackwardCompatibilityCheck/issues/143

### PR DESCRIPTION
**Pull request by Ocramius**
Fixes #134

### CODE REVIEW

File: test/unit/DetectChanges/BCBreak/PropertyBased/PropertyDocumentedTypeChangedTest.php [hunk 1]
```diff
@@ -196,6 +206,7 @@ class TheClass {
             'privateDocblockToDifferentDocblock'                                => ['[BC] CHANGED: Type documentation for property TheClass#$privateDocblockToDifferentDocblock changed from int to float'],
             'duplicatePropertyTypesBeingDeduplicatedAreNotBcBreaks'             => [],
             'propertyTypeBeingDuplicatedAreNotBcBreaks'                         => [],
+            'propertyWithComplexDocblockThatCannotBeParsed'                     => [],
```
Comment (Ocramius) [inline test/unit/DetectChanges/BCBreak/PropertyBased/PropertyDocumentedTypeChangedTest.php]: Currently, I'd say this is `mixed` -> `mixed`, so we shouldn't detect a BC break here.

File: src/DetectChanges/BCBreak/PropertyBased/PropertyDocumentedTypeChanged.php [hunk 2]
```diff
@@ -34,8 +35,13 @@ public function __invoke(ReflectionProperty $fromProperty, ReflectionProperty $t
             return Changes::empty();
         }
 
-        $fromTypes = array_unique($fromProperty->getDocBlockTypeStrings());
-        $toTypes   = array_unique($toProperty->getDocBlockTypeStrings());
+        try {
+            $fromTypes = array_unique($fromProperty->getDocBlockTypeStrings());
+            $toTypes   = array_unique($toProperty->getDocBlockTypeStrings());
+        } catch (InvalidArgumentException $failedToParseDocblock) {
+            // @TODO #134 improve docblock parsing upstream to remove this generic try-catch
```
Comment (asgrim) [inline src/DetectChanges/BCBreak/PropertyBased/PropertyDocumentedTypeChanged.php]: Not ideal to just hide it, but indeed we can't do much until we fix upstream, so this is kinda OK.

File: src/DetectChanges/BCBreak/PropertyBased/PropertyDocumentedTypeChanged.php [hunk 3]
```diff
@@ -34,8 +35,13 @@ public function __invoke(ReflectionProperty $fromProperty, ReflectionProperty $t
...
+            return Changes::empty();
```
Comment (asgrim) [inline src/DetectChanges/BCBreak/PropertyBased/PropertyDocumentedTypeChanged.php]: Do we have a way of returning "info" at all? or is it all just BC errors only?

Comment (Ocramius) [reply src/DetectChanges/BCBreak/PropertyBased/PropertyDocumentedTypeChanged.php]: I can return something, but then the build would be exiting with `1` here

Comment (asgrim) [reply src/DetectChanges/BCBreak/PropertyBased/PropertyDocumentedTypeChanged.php]: yeah, don't worry then :)

**Pull request merged**